src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/LengthConverter.cs

```diff
@@ -181,60 +181,41 @@ public override bool CanConvertTo(ITypeDescriptorContext typeDescriptorContext,
         // NOTE - This code is called from FontSizeConverter, so changes will affect both.
         static internal double FromString(string s, CultureInfo cultureInfo)
         {
-            string valueString = s.Trim();
-            string goodString = valueString.ToLowerInvariant();
-            int strLen = goodString.Length;
-            int strLenUnit = 0;
+            ReadOnlySpan<char> valueSpan = s.AsSpan().Trim();
             double unitFactor = 1.0;
 
             //Auto is represented and Double.NaN
             //properties that do not want Auto and NaN to be in their ligit values,
             //should disallow NaN in validation callbacks (same goes for negative values)
-            if (goodString == "auto") return Double.NaN;
-
-            for (int i = 0; i < PixelUnitStrings.Length; i++)
+            if (valueSpan.Equals("auto", StringComparison.OrdinalIgnoreCase))
```

Comment:
Should use SequenceEquals with an invariant comparer, supposed to be faster than Equals.

Reply:
Can you include what you think this line of code should be ? I believe this call is the most efficient way to compare 2 ReadOnlySpan<char> while using OrdinalIgnoreCase comparison.

Reply:
Actually, in my micro measurement, even ToLowerInvariant().Equals("auto") was faster than (valueSpan.Equals("auto", StringComparison.OrdinalIgnoreCase)), no idea why.

Reply:
In your micro measurement, do you do `"auto".ToLowerInvariant().Equals("auto")` ? Because in that case ToLowerInvariant does not allocate a new string because "auto" is already lowercase and in that case comparing strings might be slightly faster than span because of the need to create the span.

Reply:
No, I built AuTo from characters to compare it with auto.

Reply:
We can build a 16 element static rainbow table too that we can statically compare the binary memory bytes for absolute performance gain.

Reply:
I tried benchmarking this and I found using span to be much faster.

<details>
  <summary>Benchmark code</summary>
  
  ```csharp
[MemoryDiagnoser]
public class Benchmark1
{
    public IEnumerable<object[]> GetArguments()
    {
        yield return new object[] { "AuTo" };
    }

    [Benchmark(Baseline = true)]
    [ArgumentsSource(nameof(GetArguments))]
    public bool Old(string str1)
    {
        return str1.ToLowerInvariant().Equals("auto");
    }

    [Benchmark]
    [ArgumentsSource(nameof(GetArguments))]
    public bool New(string str1)
    {
        return str1.AsSpan().Equals("auto", StringComparison.OrdinalIgnoreCase);
    }
}
  ```
  
</details>

<details>
  <summary>Benchmark results</summary>
  
| Method | str1 | Mean       | Error     | StdDev    | Ratio | Gen0   | Allocated | Alloc Ratio |
|------- |----- |-----------:|----------:|----------:|------:|-------:|----------:|------------:|
| Old    | AuTo | 16.6884 ns | 0.3390 ns | 0.3005 ns |  1.00 | 0.0051 |      32 B |        1.00 |
| New    | AuTo |  0.9111 ns | 0.0370 ns | 0.0309 ns |  0.05 |      - |         - |        0.00 |
  
</details>

Reply:
Can you try AsSpan().ToLowerInvariant()?

Reply:
@psmulovics Sorry, I do not think `AsSpan().ToLowerInvariant()` has any advantage.

Reply:
I am still slightly concerned on the 15% slower 'auto' - as mentioned above, can we do a rainbow table of the 16 possible ways to write auto and compare it bitwise?

Reply:
I forgot to mention it but I opened an issue in dotnet/runtime for the regression of "auto": dotnet/runtime#96524. It is a performance regression introduced in .Net 8. I have a workaround that I'll post soon which should remove the slight regression of "auto".

> I am still slightly concerned on the 15% slower 'auto' - as mentioned above, can we do a rainbow table of the 16 possible ways to write auto and compare it bitwise?

String comparisons are already highly optimized and we shouldn't make the code way more complex just to maybe make a ~8 ns case a bit faster.

Reply:
I pushed a commit that fixed the perf regression and I updated the benchmark results. Every case improved now, I'm marking this conversation as resolved.